server/game/cards/05-DT/SelectivePreservation.js

```diff
@@ -3,20 +3,44 @@ const Card = require('../../Card.js');
 class SelectivePreservation extends Card {
     setupCardAbilities(ability) {
         this.play({
-            target: {
-                activePromptTitle: 'Choose a creature of each power value to not destroy',
-                cardType: 'creature',
-                mode: 'exactly',
-                numCards: (context) =>
-                    new Set(context.game.creaturesInPlay.map((card) => card.power)).size,
-                selectorCondition: (selectedCards, context) =>
-                    new Set(context.game.creaturesInPlay.map((card) => card.power)).size ===
-                    new Set(selectedCards.map((card) => card.power)).size,
-                gameAction: ability.actions.destroy((context) => ({
-                    target: context.game.creaturesInPlay.filter(
-                        (card) => context.target && !context.target.includes(card)
+            effect: 'Select creatures for each power to not destroy.',
+
+            then: (preThenContext) => {
+                let unqiuePowers = Array.from(
```

Comment:
I believe the preference in the code for unique is to use _.uniq

Reply:
the preference is to never use underscore where possible :)

Reply:
ops, my bad! I have been using, so I'll try to avoid.